tests/HelperFunctions.py

```diff
@@ -0,0 +1,60 @@
+# ===================================================
+#
+#    Copyright (c) 2023-2024
+#      SPARKX Team
+#
+#    GNU General Public License (GPLv3 or later)
+#
+# ===================================================
+
+import math
+
+"""
+This module contains helper functions and global variables that are used in 
+the testing scripts.
+"""
+
+# Global variables
+small_value = 1e-9
+
+
+# Helper functions
+
+# Converts a string to a float or an integer, if applicable.
+# Otherwise, returns the original item.
+def convert_str_to_number(item):
+    try:
+        if '.' in item or 'e' in item or 'E' in item:
+            return float(item)
+        else:
+            return int(item)
+    except ValueError:
+        return item
+
+
+# Converts all string elements in a nested list to their respective
+# numerical types.
+def convert_nested_list_to_numerical(nested_list):
+    return [[convert_str_to_number(item) for item in sublist] for sublist in nested_list]
+
+# Compares two floating-point numbers within a given tolerance.
+def compare_floats(a, b, tol=small_value):
```

Comment:
I am almost sure there is a pytest functionality for this, and potentially many of the functions here.

Reply:
> I am almost sure there is a pytest functionality for this, and potentially many of the functions here.

I have deleted the compare_floats() method and integrated it into compare_nested_lists() using the pytest syntax.
As far a I see, there is no pytest functionality that compares two nested lists. However, I have also added tests for the HelperFunctions to capture potential bugs (even though I know that a comment might be that we shouldn't test tests, I think it makes sense here, as HelperFunctions is basically a collection of free utility functions)